### PR TITLE
Feat :  Comment count in /api/articles/article_id

### PR DESCRIPTION
**Pull request by ji0kim**
GET  /api/articles/article_id
Show comment_count when responding with an object of article

### CODE REVIEW

File: models/articles.model.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const db = require('../db/connection');
 const format = require('pg-format');
+const { lastIndexOf } = require('../db/data/test-data/articles');
```
Comment (dentednerd) [inline models/articles.model.js]: What's going on here?? Could you please explain for me?

File: models/articles.model.js [hunk 2]
```diff
@@ -10,14 +11,23 @@ exports.selectArticles = () => {
...
+	return db
+		.query(queryTxt, [article_id])
+		.then((result) => {
+			if (result.rows.length === 0) {
+				return Promise.reject({ status: 404, msg: 'Not found' });
+			}
+			return result;
+		})
+		.then((result) => {
+			return db.query(commentCountQueryTxt, [article_id]).then((commentById) => {
```
Comment (dentednerd) [inline models/articles.model.js]: You don't need to do two queries here. There's a much more succinct way of getting the comment count.

**Review by dentednerd — CHANGES_REQUESTED** [review]
Some refactoring needs to happen here Jiyoung!

File: models/articles.model.js (hunk 1 above)
Comment (ji0kim) [reply models/articles.model.js]: @dentednerd  Oops I think it's automatically injected whilst I was saving..(?) I'll remove this.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -171,22 +176,44 @@ describe('GET /api/users', () => {
...
+					title: 'Living in the shadow of a great man',
+					topic: 'mitch',
+					author: 'butter_bridge',
+					created_at: '2020-07-09T20:11:00.000Z',
+					body: 'I find this existence challenging',
+					votes: 100,
+					comment_count: 11,
+				});
+			});
+	});
```
Comment (jadelandeg) [inline __tests__/app.test.js:218]: Nice test! Can also check that selecting an article with no comments gives a 200.

Need some error testing here, 404 for article not found and 400 for invalid article id :)

File: models/articles.model.js [hunk 4]
```diff
@@ -2,21 +2,26 @@ const db = require('../db/connection');
...
 exports.selectArticleById = (article_id) => {
-	const queryTxt = format('SELECT * FROM articles WHERE article_id = $1');
+  const queryTxt = format(`
+    SELECT articles.*, COUNT(comment_id)::INT AS comment_count
+    FROM articles
+    LEFT JOIN comments
+    ON articles.article_id = comments.article_id 
+    WHERE articles.article_id = $1
+    GROUP BY articles.article_id;
+  `);
```
Comment (jadelandeg) [inline models/articles.model.js:19]: Nice query, Jiyoung! :)

**Review by jadelandeg — CHANGES_REQUESTED** [review]
Nice work on the query! Smashing it. Just a few more tests for error handling :)

File: __tests__/app.test.js (hunk 3 above)
Comment (ji0kim) [reply __tests__/app.test.js:218]: @jadelandeg 
Hi Jade! Thank you for the comment!

I noticed while writing the code that from the endpoints below.
- GET '/api/articles/article_id' 
- PATCH '/api/articles/article_id' 

These 2 have 404, 400 test for article_id  already. Is it better to include 400, 404 err testing code in this one to be more explicit?

**Review by jadelandeg — APPROVED** [review]
Nice work! Ignore previous comments about error handling, didn't realise it was the same endpoint and have just added comment count. The errors have already been handled.

**Pull request merged**